docs: update missing suspense documentation, add debugging steps (#80448)
vercel/next.js: errors/missing-suspense-with-csr-bailout.mdx

Reading search parameters through `useSearchParams()` without a Suspense boundary will opt the entire page into client-side rendering. This could cause your page to be blank until the client-side JavaScript has loaded.

> Common scenarios include:
> Directly calling `useSearchParams()` in a page or component without wrapping it in `<Suspense>`.
> 
> - A provider (e.g., a query or context provider) in the component tree using `useSearchParams()` without proper `<Suspense>` wrapping, triggering the CSR bailout indirectly.
> - A deeply nested component that uses `useSearchParams()` without being wrapped in a Suspense boundary, causing the entire page to be rendered on the client side.
> 

Reviewer: The first and second are similar as they both mention component, can delete. 

```suggestion
Common scenarios include:
- Directly calling `useSearchParams()` in a page or nested component without wrapping it in `<Suspense>`.
- A provider (e.g. a query or context provider) in the component tree using `useSearchParams()` without `<Suspense>, triggering the CSR bailout indirectly.
```

This will ensure the page does not de-opt to client-side rendering.

## Debugging Steps

If you encounter a CSR bailout but cannot identify the source, the issue may stem from a provider or parent component rather than the page itself. Follow these steps to debug:
### 1. Inspect the Component Tree:
- Review your page and layout components for direct calls to `useSearchParams()`.
- Check for providers (e.g. `QueryClientProvider`, custom context providers and external packages like `nuqs`) that might internally call `useSearchParams()`.
- Use React Developer Tools to inspect the component hierarchy and identify where `useSearchParams()` is used.
- Run a global search in your codebase for `useSearchParams()` and check that each usage is wrapped in a `<Suspense>` boundary.

### 2. Add `<Suspense>` incrementally

> - Temporarily wrap individual components or providers in `<Suspense>` boundaries to isolate the components.

Author: imo 'isolate' conveys that as well

This will ensure the page does not de-opt to client-side rendering.

> ## Debugging Steps
> If you encounter a CSR bailout but cannot identify the source, the issue may stem from a provider or parent component rather than the page itself. Follow these steps to debug:

Reviewer: ```suggestion
## Debugging Steps

If you encounter a CSR bailout but cannot identify the source, the issue may stem from a provider or parent component rather than the page itself. Follow these steps to debug:
```

> ### 1. Check for the Error Message: 
> - Look for the Next.js warning in the console, such as A client component is reading search params without a Suspense boundary. This indicates a CSR bailout.

Reviewer: If they land on this error message page that means they likely already encountered the error, can delete. 

```suggestion
```

> ### 2. Inspect the Component Tree:
> - Review your page and layout components for direct calls to `useSearchParams()`.
> - Check for providers (e.g., `QueryClientProvider`, custom context providers) that might internally call `useSearchParams()`.
> - Use React Developer Tools to inspect the component hierarchy and identify where `useSearchParams()` is used.

Reviewer: ```suggestion
### 1. Inspect the Component Tree:

- Review your page and layout components for direct calls to `useSearchParams()`.
- Check for providers (e.g. `QueryClientProvider`, custom context providers) that might internally call `useSearchParams()`.
- Use React Developer Tools to inspect the component hierarchy and identify where `useSearchParams()` is used.
- Run a global search in your codebase for `useSearchParams()` and check that each usage is wrapped in a `<Suspense>` boundary.
```

> ### 3. Search for `useSearchParams` Usage:
> - Run a global search in your codebase for `useSearchParams` to locate all instances.
> - Verify that each usage is wrapped in a `<Suspense>` boundary.

Reviewer: This step is similar to 2, can delete. 

```suggestion
```

> ### 4. Test with Incremental Suspense:
> - Temporarily wrap individual components or providers in `<Suspense>` boundaries to isolate the problematic component.
> - Example: If a provider in layout.tsx is suspected, wrap it in `<Suspense>` and observe if the bailout resolves.

Reviewer: ```suggestion
### 2. Add `<Suspense>` incrementally

- Temporarily wrap individual components or providers in `<Suspense>` boundaries to isolate the components.
- Example: If a provider in layout.tsx is suspected, wrap it in `<Suspense>` and observe if the bailout resolves.
```

> ### 5. Check Third-Party Libraries:
> - Some libraries (e.g., query or state management libraries like `nuqs`) may internally use `useSearchParams()`. Check their documentation or source code.
> - Ensure the library’s provider is wrapped in `<Suspense>` if it relies on `useSearchParams()`.

Reviewer: Again, similar to step 1, can delete. 

```suggestion
```
